src/com/google/enterprise/adaptor/UrlValidator.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.adaptor;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Validates URLs by syntax checking and validating the host is reachable.
+ * No attempts to fetch the URL are made, so a mocked up URL with
+ * placeholder values in format substitutions may be used.
```

Comment:
Hmm. That wasn't my intended usage. I thought that connectors should plug in one of the start paths in init. Are unescaped braces legal in an URL? I think that checking for unsubstituted braces would be a good idea, as a common error: e.g., using {2} when there are just 2 possible parameters.